src/sentry_value.c

```diff
@@ -1124,10 +1124,32 @@ sentry_value_new_stacktrace(void **ips, size_t len)
     return stacktrace;
 }
 
+sentry_value_t
+sentry__value_new_span(sentry_value_t parent, const char *operation)
+{
+    sentry_value_t span = sentry_value_new_object();
+
+    sentry_transaction_context_set_operation(span, operation);
+    sentry_value_set_by_key(span, "status", sentry_value_new_string("ok"));
+
+    if (!sentry_value_is_null(parent)) {
+        sentry_value_set_by_key(span, "trace_id",
+            sentry_value_get_by_key_owned(parent, "trace_id"));
+        sentry_value_set_by_key(span, "parent_span_id",
+            sentry_value_get_by_key_owned(parent, "span_id"));
+        sentry_value_set_by_key(
+            span, "sampled", sentry_value_get_by_key_owned(parent, "sampled"));
+    }
+
+    return span;
+}
+
 sentry_value_t
 sentry_value_new_transaction_context(const char *name, const char *operation)
 {
-    sentry_value_t transaction_context = sentry_value_new_object();
+    sentry_value_t transaction_context
+        = sentry__value_new_span(sentry_value_new_null(), operation);
```

Comment:
On line 1163 (not part of diff so i can't comment there) the operation is set again.  this is now not needed.

Reply:
ah, thanks for catching this